tensorflow/lite/micro/test_helpers.cc

```diff
@@ -193,10 +195,17 @@ const Model* ModelBuilder::BuildModel(
 
   // TFLM only supports single subgraph.
   constexpr size_t subgraphs_size = 1;
+
+  // Find out number of subgraph inputs.
+  int num_tensors_in_subgraph_inputs = inputs.size(); // Default case = all inputs
+  if (num_subgraph_inputs > 0 && num_subgraph_inputs < inputs.size()) {
```

Comment:
I'd prefer having the separation between the default case and the ethos-u case be more explicit, and avoiding the need for `num_tensors_in_subgraph_inputs`.

Something like:

```cpp

if (num_subgraph_inputs == 0) {
  // this is the default case (i.e. for all platforms other than ethos-u)
  num_subgraph_inputs = inputs.size()
} else {
  // for ethos-u, a non-zero value of num_subgraph_inputs means that some of the inputs point to scratch buffers.   
  TF_LITE_DCHECK(num_subgraph_inputs < inputs.size());
}
```
and then this line can be:
```
 builder_->CreateVector(inputs.begin(), num_subgraph_inputs)
```

Reply:
Done